### ISSUE TITLE
webhook remote error: tls: bad certificate

### ISSUE THREAD

**Issue by andresm53** [external-secrets/external-secrets#2271]
Environment: Openshift 4.12.3

I simply installed external-secrets by running:

helm install external-secrets \
   external-secrets/external-secrets \
    -n external-secrets \
    --create-namespace

Tried to sync a Secret from AWS Secret Manager, works ok.

However, I see that the external-secrets-webhook is constantly throwing this error:
`2023/04/28 18:31:37 http: TLS handshake error from 10.130.0.34:54888: remote error: tls: bad certificate`

Where 10.130.0.34 in this case is kube-apiserver-operator, which at the same time is throwing:
`E0428 18:36:38.059056 1 degraded_webhook.go:128] x509: certificate signed by unknown authority`

I have reviewed other issues (already closed) related to the webhook error "bad certificate" (for example [this](https://github.com/external-secrets/external-secrets/issues/997)), but those don't seem to be related to this one. In this case, this is a new Openshift cluster with no workloads running and ExternalSecret was just installed with latest version and the error comes up immediately.

EDIT: I just noticed that helm install throwed these warnings, not sure if this could be related to the problem:

```
W0428 15:28:12.580724     651 warnings.go:70] would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "webhook" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "webhook" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "webhook" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "webhook" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")

W0428 15:28:12.580725     651 warnings.go:70] would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "external-secrets" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "external-secrets" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "external-secrets" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "external-secrets" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")

W0428 15:28:12.583534     651 warnings.go:70] would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "cert-controller" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "cert-controller" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "cert-controller" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "cert-controller" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

Any ideas on how to fix these certificate errors? 

Thanks!

**Comment by gusfcarvalho** [comment]
Hi @andresm53 ! Did you try to install with the community operators package? I believe in openshift you need to provide extra configuration/ permissions that you can’t with the default helm install.

to troubleshoot the issue, a secret called external-secrets-Webhook should be created with a certificate configuration. Also, the CRDs should have been patched to have that CA in place.

You want to look at webhook and certController pods

**Comment by andresm53** [comment]
Hi @gusfcarvalho 
Thanks. You were right. Instead of installing external-secrets directly through helm, I tried this:
- Went to the Openshift console > OperatorHub and installed external-secrets from there.
- Then I created an Operator Config ([sample](https://github.com/external-secrets/external-secrets-helm-operator/blob/main/config/samples/operator_v1alpha1_operatorconfig.yaml)).
And now it worked ok (no more certificate errors).
Thanks for you help.
